src/libsyntax/feature_gate/check.rs

```diff
@@ -766,12 +769,16 @@ pub fn get_features(span_handler: &Handler, krate_attrs: &[ast::Attribute],
                     continue;
                 }
 
-                for &(name, .., f_edition, set) in ACTIVE_FEATURES {
+                for &Feature { name, edition: f_edition, state, .. } in ACTIVE_FEATURES {
                     if let Some(f_edition) = f_edition {
                         if f_edition <= *edition {
```

Comment:
...Same here, except `crate_edition` substituted for `edition`. Could use a function for these, e.g. `fn active_features_until_edition(edition: ...)`.